dask/array/routines.py

```diff
@@ -332,6 +333,52 @@ def vdot(a, b):
     return dot(a.conj().ravel(), b.ravel())
 
 
+def _shape_minus_axis(shape, axis):
+    newshape = list(shape)
+    newshape.pop(axis)
+    return tuple(newshape)
+
+
+def _chunk_sum(a, axis=None, dtype=None, keepdims=None):
+    # Caution: this is not your conventional array-sum: due
+    # to the special nature of the preceding blockwise con-
+    # traction,  each chunk is expected to have exactly the
+    # same shape,  with a size of 1 for the dimension given
+    # by `axis` (the reduction axis).  This makes mere ele-
+    # ment-wise addition of the arrays possible.   Besides,
+    # the output can be merely reshaped to lose the `axis`-
+    # dimension when keepdims = False
+    if type(a) is list:
+        xp = np
+
+        if is_cupy_type(a[0]):
+            import cupy
+
+            xp = cupy
+
+        out = reduce(partial(xp.add, dtype=dtype), a)
+    else:
+        if a.shape == (0,):
+            return a
+        out = a
+    if keepdims:
+        return out
+    else:
+        return out.reshape(_shape_minus_axis(out.shape, axis[0]))
```

Comment:
We can probably make the squeeze work (see the other comment), but otherwise if this method is used only once, probably not worth a complete new method.

Reply:
I agree.  Let me know, after you read my response to adding a squeeze method to `EncapsulatedNDArray`, what you think.